File: moody_study/lib/models/generated_quiz_response.dart
class GeneratedQuizResponse {
  final int id;
  final int materialId;
  final String fileName;
  final String quizContent;
  final String generatedAt;
  final bool saved;

  GeneratedQuizResponse({
    required this.id,
    required this.materialId,
    required this.fileName,
    required this.quizContent,
    required this.generatedAt,
    this.saved = false,
  });

  factory GeneratedQuizResponse.fromJson(Map<String, dynamic> json) {
    return GeneratedQuizResponse(
      id: (json['id'] as num).toInt(),
      materialId: (json['materialId'] as num).toInt(),
      fileName: json['fileName'] as String,
      quizContent: json['quizContent'] as String,
      generatedAt: json['generatedAt'] as String,
      saved: json['saved'] as bool? ?? false,
    );
  }
}

File: moody-study-backend/src/main/java/com/example/moody_study_backend/dto/GeneratedQuizResponse.java
package com.example.moody_study_backend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class GeneratedQuizResponse {
    private Long id;
    private Long materialId;
    private String fileName;
    private String quizContent;
    private String generatedAt;
    private boolean saved;
}